services/common/src/components/project/ProjectDescriptionTab.tsx

```diff
@@ -0,0 +1,472 @@
+      const authsOfType = authorizations[type];
+      if (authsOfType) {
+        if (isArray(authsOfType)) {
+          const formattedAuthorizations = authsOfType.map((a) => {
+            return transformAuthorization(type, a);
+          });
+          updatedAuthorizations = updatedAuthorizations.concat(formattedAuthorizations);
+        } else {
+          newAmsAuthorizations = newAmsAuthorizations.concat(
+            authsOfType?.NEW.map((a) =>
+              transformAuthorization(type, {
+                ...a,
+                project_summary_permit_type: ["NEW"],
+              })
+            )
+          );
+          amendAmsAuthorizations = amendAmsAuthorizations.concat(
+            authsOfType?.AMENDMENT.map((a) =>
+              transformAuthorization(type, {
+                ...a,
+                project_summary_permit_type: ["AMENDMENT"],
+              })
+            )
+          );
+        }
+      }
+    });
+    return {
+      authorizations: updatedAuthorizations,
+      ams_authorizations: { amendments: amendAmsAuthorizations, new: newAmsAuthorizations },
+    };
+  };
+
+  const handleTransformPayload = (payload: any) => {
+    let payloadValues: any = {};
+    const updatedAuthorizations = transformAuthorizations(payload);
+    const values = removeNullValuesRecursive(payload);
+    payloadValues = {
+      ...values,
+      ...updatedAuthorizations,
+    };
+    delete payloadValues.authorizationTypes;
+    return payloadValues;
+  };
+
+  /* Transforms project summary authorizations to match the
+   *  shape of project summary authorization form values.
+   */
+  const transformProjectSummaryAuthorizations = (input) => {
+    const output: any = {};
+    input.forEach((authorization) => {
+      const authType = authorization.project_summary_authorization_type;
+
+      if (!output[authType]) {
+        output[authType] = { types: [], AMENDMENT: [], NEW: [] };
+      }
+
+      authorization.project_summary_permit_type.forEach((permitType) => {
+        if (!output[authType].types.includes(permitType)) {
+          output[authType].types.push(permitType);
+        }
+        if (permitType === "AMENDMENT") {
+          if (!authorization.amendment_changes) {
+            authorization.amendment_changes = [];
+          }
+          output[authType].AMENDMENT.push(authorization);
+          if (!output[authType].NEW) {
+            output[authType].NEW = [];
+          }
+        } else if (permitType === "NEW") {
+          output[authType].NEW.push(authorization);
+          if (!output[authType].AMENDMENT) {
+            output[authType].AMENDMENT = [];
+          }
+        }
+      });
+    });
+
+    if (output.MINES_ACT_PERMIT) {
+      output.MINES_ACT_PERMIT = input.filter(
+        (auth) => auth.project_summary_authorization_type === "MINES_ACT_PERMIT"
+      );
+    }
+
+    return output;
+  };
+
+  const handleRetryAMSSubmissionClicked = async () => {
```

Comment:
There seems to be a lot of logic here just to retry the submission to AMS. Any reason not to set up an endpoint in the API and re-trigger the jobs with the data we already have?